elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -139,4 +139,54 @@ default void flush() {
                                                           Optional<Sorting> sorting, Optional<Pagination> pagination) {
         return collection;
     }
+
+    default <T> Object getRelation(
+            Object entity,
+            RelationshipType relationshipType,
+            String relationName,
+            Class<T> relationClass,
+            EntityDictionary dictionary,
+            Set<Predicate> filters
+    ) {
+        Object val = PersistentResource.getValue(entity, relationName, dictionary);
+        if (val instanceof Collection) {
+            Collection filteredVal = (Collection) val;
+
+            if (!filters.isEmpty()) {
+                filteredVal = filterCollection(filteredVal, relationClass, filters);
+            }
+            return filteredVal;
+        }
+
+        return val;
+    }
+
+    default <T> Object getRelationWithSortingAndPagination(
+            Object entity,
+            RelationshipType relationshipType,
+            String relationName,
+            Class<T> relationClass,
+            EntityDictionary dictionary,
+            Set<Predicate> filters,
+            Sorting sorting,
+            Pagination pagination
+    ) {
+        Object val = PersistentResource.getValue(entity, relationName, dictionary);
+        if (val instanceof Collection) {
+            Collection filteredVal = (Collection) val;
+            // sorting/pagination supported on last entity only eg /v1/author/1/books? books would be valid
+            final boolean hasSortRules = sorting.isDefaultInstance();
+            final boolean isPaginated = pagination.isDefaultInstance();
+
+            if (!filters.isEmpty() || hasSortRules || isPaginated) {
+                final Optional<Sorting> sortingRules = hasSortRules ? Optional.of(sorting) : Optional.empty();
+                final Optional<Pagination> paginationRules = isPaginated ? Optional.of(pagination) : Optional.empty();
+                filteredVal = filterCollectionWithSortingAndPagination(filteredVal,
```

Comment:
Let's deprecate filterCollection and filterCollectionWithSortingAndPagination.  That functionality should move into the specific implementation of the data store transactions.